core/base/discreteMorseSandwich/DiscreteMorseSandwich.h

```diff
@@ -433,25 +433,58 @@ namespace ttk {
       if(dim > 3 || dim < 1) {
         return;
       }
-      this->firstRepMin_.resize(triangulation.getNumberOfVertices());
-      if(dim > 1) {
-        this->firstRepMax_.resize(triangulation.getNumberOfCells());
-      }
-      if(dim > 2) {
-        this->critEdges_.resize(triangulation.getNumberOfEdges());
-        this->edgeTrianglePartner_.resize(triangulation.getNumberOfEdges(), -1);
-        this->onBoundary_.resize(triangulation.getNumberOfEdges(), false);
-        this->s2Mapping_.resize(triangulation.getNumberOfTriangles(), -1);
-        this->s1Mapping_.resize(triangulation.getNumberOfEdges(), -1);
-      }
-      for(int i = 0; i < dim + 1; ++i) {
-        this->pairedCritCells_[i].resize(
-          this->dg_.getNumberOfCells(i, triangulation), false);
-      }
-      for(int i = 1; i < dim + 1; ++i) {
-        this->critCellsOrder_[i].resize(
-          this->dg_.getNumberOfCells(i, triangulation), -1);
+#ifdef TTK_ENABLE_OPENMP
+#pragma omp parallel master num_threads(threadNumber_)
```

Comment:
I'd put the `#endif` right here. then you don't need the `#ifdef` at the very end.